scripts/util/doc-builder.js

```diff
@@ -0,0 +1,236 @@
+#!/usr/bin/env node
+
+var path = require('path');
+var fs = require('fs');
+var yuidoc = require('./yuidoc');
+var NodeGit = require('nodegit');
+var util = require('util');
+var fsExtra = require('fs-extra');
+
+function DocBuilder(options) {
```

Comment:
Yeah, I really like the factory approach, and have no problem with that in general (although using an uppercase name for the factory makes it a little more confusing, I'd assume it's going to be a constructor function in that case). Much of the JS/ CoffeeScript in the rest of Shopify is much more class-oriented, though, which largely comes from it being done by people who were more comfortable writing Ruby.